src/hooks/users.ts

```diff
@@ -1,15 +1,97 @@
 
-import { usersSelector, usersByIdSelector } from "utils/selectors";
+import { universeUsersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
 import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
+import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
 
 import { User } from "types/User";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+
+export const useRecentPartyUsers = () => [];
+// export const useRecentPartyUsers = (venueName: string): WithId<User>[] => {
+//   const venueId = useVenueId();
+//   const partygoers = useSelector(partygoersSelector) ?? [];
+
+//   const { parentVenue, relatedVenues } = useConnectRelatedVenues({
+//     venueId,
+//   });
+
+//   const lastSeenThreshold = useUserLastSeenThreshold();
+
+//   const parentVenueName = parentVenue?.name ?? "";
+
+//   const filterPartygoersByHierarchy = useCallback(
+//     (partygoer: WithId<User>) => {
+//       return relatedVenues?.map((venue) => {
+//         return (
+//           partygoer?.lastSeenIn?.[parentVenueName || venueName || venue.name] >
+//           lastSeenThreshold
+//         );
+//       });
+//     },
+//     [lastSeenThreshold, parentVenueName, relatedVenues, venueName]
+//   );
+
+//   return useMemo(() => partygoers.filter(filterPartygoersByHierarchy), [
+//     partygoers,
+//     filterPartygoersByHierarchy,
+//   ]);
+// };
+
+// NOTE: Heuristic approach. Way faster and efficient.
+// Is based on assumption that there is only one parent venue and it is an entry point for any user
+export const useConnectUniverseUsers = () => {
+  const venueId = useVenueId();
+
+  const { currentVenue, isCurrentVenueLoaded } = useConnectCurrentVenueNG(
+    venueId
+  );
+
+  useSparkleFirestoreConnect(() => {
+    if (!isCurrentVenueLoaded || !currentVenue) return [];
+
+    const rawVenueId = currentVenue.parentId || currentVenue.name;
+    const venueId = rawVenueId.toLowerCase();
+    return [
+      {
+        collection: "users",
+        where: ["enteredVenueIds", "array-contains", venueId],
+        storeAs: "universeUsers",
+      },
+    ];
+  });
+};
+
+// NOTE: Logically correct version of pulling users
+export const useConnectWorldUsers_V2 = () => {
+  const venueId = useVenueId();
+
+  const { relatedVenues, isRelatedVenuesLoaded } = useConnectRelatedVenues({
+    venueId,
+  });
+
+  useFirestoreConnect(() => {
+    if (!isRelatedVenuesLoaded) return [];
+
+    const worldUsersVenueNames = relatedVenues.map((venue) =>
+      venue.name.toLowerCase()
+    );
+
+    console.log({ worldUsersVenueNames });
+
+    return worldUsersVenueNames.map((venueName) => ({
+      collection: "users",
+      where: ["enteredVenueIds", "array-contains", venueName],
```

Comment:
Does `enteredVenueIds` actually store the `venueName` in it? Or does it store the `venueId`?

Also, how are we accounting for rooms, as I think they may also get stored in `enteredVenueIds` (if I remember correctly)

Reply:
As far as I could get from the field research, the difference between venueName vs venueId is casing. `enteredVenueIds` stores Ids. I think you are referring to the naming with such questions. Will fix that in my next updates.

I didn't know rooms could be stored in `enteredVenueIds`  ??? If this is true, then we would need to change the logic of this firestore field. Because its naming is quite explicit about what is stored there.

Reply:
> I think you are referring to the naming with such questions. Will fix that in my next updates.

In the code snippet here you have `where: ["enteredVenueIds", "array-contains", venueName]`, whereas I expect what we actually want that to be is where: ["enteredVenueIds", "array-contains", venueId]`

> I didn't know rooms could be stored in `enteredVenueIds`?

I may be misremembering / mixing up `enteredVenueIds` with the general 'counting' fields, so you may be correct that this field can only store `venueId`s